guard/src/rules/functions/strings.rs

```diff
@@ -177,13 +177,23 @@ pub(crate) fn join(args: &[QueryResult], delimiter: &str) -> crate::rules::Resul
             }
             QueryResult::UnResolved(ur) => {
                 return Err(Error::IncompatibleError(format!(
-                    "Joining non unresolved values is not allowed {}, unsatisfied part {}",
+                    "Joining unresolved values is not allowed {}, unsatisfied part {}",
                     ur.traversed_to, ur.remaining_query
                 )));
             }
         }
     }
-    Ok(PathAwareValue::String((Path::root(), aggr)))
+
+    match args.is_empty() {
```

Comment:
Should we add an integration test for this like I did with count yesterday?

Reply:
good catch, I was planning to, but over the course of raising the PR I forgot to.